bulla-contracts/src/mappings/FrendLend.ts

```diff
@@ -97,3 +184,82 @@ export function handleLoanOfferRejected(event: LoanOfferRejected): void {
   user_creditor.save();
   user_debtor.save();
 }
+
+export function handleFeeWithdrawn(event: FeeWithdrawn): void {
+  const ev = event.params;
+
+  const feeWithdrawnEvent = createFeeWithdrawnEvent(event);
+  const user_admin = getOrCreateUser(ev.admin);
+
+  feeWithdrawnEvent.admin = ev.admin;
+  feeWithdrawnEvent.token = getOrCreateToken(ev.token).id;
+  feeWithdrawnEvent.amount = ev.amount;
+
+  feeWithdrawnEvent.eventName = "FeeWithdrawn";
+  feeWithdrawnEvent.blockNumber = event.block.number;
+  feeWithdrawnEvent.transactionHash = event.transaction.hash;
+  feeWithdrawnEvent.logIndex = event.logIndex;
+  feeWithdrawnEvent.timestamp = event.block.timestamp;
+
+  user_admin.frendLendEvents = user_admin.frendLendEvents ? user_admin.frendLendEvents.concat([feeWithdrawnEvent.id]) : [feeWithdrawnEvent.id];
+
+  feeWithdrawnEvent.save();
+  user_admin.save();
+}
+
+export function handleLoanPayment(event: LoanPayment): void {
+  const ev = event.params;
+
+  const loanPaymentEvent = createLoanPaymentEvent(event);
+
+  loanPaymentEvent.claimId = ev.claimId.toString();
+  loanPaymentEvent.grossInterestPaid = ev.grossInterestPaid;
+  loanPaymentEvent.principalPaid = ev.principalPaid;
+  loanPaymentEvent.protocolFee = ev.protocolFee;
+
+  loanPaymentEvent.eventName = "LoanPayment";
+  loanPaymentEvent.blockNumber = event.block.number;
+  loanPaymentEvent.transactionHash = event.transaction.hash;
+  loanPaymentEvent.logIndex = event.logIndex;
+  loanPaymentEvent.timestamp = event.block.timestamp;
+
+  loanPaymentEvent.save();
+
+  // Update the underlying claim that was created when the loan was accepted
+  const claim = getOrCreateClaim(ev.claimId.toString());
+  const newPaidAmount = claim.paidAmount.plus(ev.principalPaid);
+  const isClaimPaid = newPaidAmount.ge(claim.amount);
+
+  claim.paidAmount = newPaidAmount;
```

Comment:
I noticed we aren't adding this in frendlend events, is that consistent with V1?

Reply:
you mean newPaidAmount as a parameter in the schema?

Reply:
no, I meant adding the event to the frendlendEvents list, just like the handler below

Reply:
needs to be added

Reply:
We would need to add it to the list, because we now have interest only payments, and interest payments in general that will only appear on this event. Doesn't matter that it doesn't exist in V1